### ISSUE TITLE
pylint is extremely slow on Generic classes

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#4079]
The following code:
`testGenerics.py`
```python3
from typing import Generic, TypeVar

T = TypeVar('T')

class A(Generic[T]): pass
class B(A[T]): pass
class C(B[T]): pass
class D(C[T]): pass
class E(D[T]): pass
class F(E[T]): pass
class G(F[T]): pass

print("OK")
```
produces the following output:
```python3
$ python3 testGenerics.py 
OK

$ mypy testGenerics.py 
Success: no issues found in 1 source file

$ time pylint testGenerics.py 
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

real	0m32.662s
user	0m32.219s
sys	0m0.233s

$ pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0]
```
Note the **32 seconds** run time.

Increasing the number of generic classes in a chain can easily bump the `pylint` run time to an hour or more.

The similar code without generics:
`testNoGenerics.py`
```python3
class A(): pass
class B(A): pass
class C(B): pass
class D(C): pass
class E(D): pass
class F(E): pass
class G(F): pass

print("OK")
```
has no such problems:
```python3
$ python3 testNoGenerics.py 
OK

$ time pylint testNoGenerics.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

real	0m1.505s
user	0m1.340s
sys	0m0.127s
```

**Comment by robin-wayve** [comment]
I'm not sure this example is still slow: I just had it pass in 2 seconds on Pylint 2.9.6 🤔
